### ISSUE TITLE
Remove unused deps from workspace Cargo.toml

### ISSUE THREAD

**Issue by utkarshgupta137** [bnjbvr/cargo-machete#79]
Currently, cargo-machete doesn't check if the workspace Cargo.toml contains any unused deps or not. I think it should be straightforward as you only need to deal with `Cargo.toml` files, although things never are.

**Comment by tisonkun** [comment]
@bnjbvr any idea on this issue?

**Comment by bnjbvr** [comment]
Yep, it's not implemented; one would need to scan the set of dependencies from the workspace, and then figure out the workspace definition to navigate through all the directories it includes. Good luck :-)

**Comment by HaoYang670** [comment]
Hi, is there any progress?

**Comment by bnjbvr** [comment]
Nope, no one has volunteered to work on this, so feel free to take a stab and open a PR!

**Comment by TD-Sky** [comment]
Removing a dependency with `cargo remove` will clean all unused dependencies in workspace `Cargo.toml`.

**Comment by HaoYang670** [comment]
> Removing a dependency with cargo remove will clean all unused dependencies in workspace Cargo.toml.

No I don't think so. `cargo remove` needs user to provide the name of dependency to remove. It cannot find unused deps.

**Comment by TD-Sky** [comment]
> > Removing a dependency with cargo remove will clean all unused dependencies in workspace Cargo.toml.
> 
> No I don't think so. `cargo remove` needs user to provide the name of dependency to remove. It cannot find unused deps.

You can give a try. That is the side effect of removing package.

**Comment by bnjbvr** [comment]
Sweet, I didn't know about `cargo-shear`! Looks like there are a few good ideas to borrow from there…
